packages/db/src/query/reports.ts

```diff
@@ -0,0 +1,167 @@
+import { and, desc, eq, gt, lte, or, sql } from "drizzle-orm";
+
+import type { DayStatistics, ReportProps } from "@energyleaf/lib";
+import db from "../";
+import { historyReportConfig, reportConfig, reports, reportsDayStatistics } from "../schema/reports";
+import { user } from "../schema/user";
+
+/**
+ * Update the user report settings data in the database
+ */
+export async function updateReportConfig(
+    data: {
+        receiveMails: boolean;
+        interval: number;
+        time: number;
+    },
+    id: string,
+) {
+    return db.transaction(async (trx) => {
+        const oldReportData = await getReportConfigByUserId(id);
```

Comment:
`getReportConfigByUserId` wird hier glaube ich nicht in der gleichen Transkation ausgeführt. Denke nicht, dass wir so jemals in Probleme laufen werden. Aber sollte man vllt doch prüfen und dann sauber machen